src/index.js

```diff
@@ -67,15 +67,28 @@ const parseDate = (cfg) => {
   if (date instanceof Date) return new Date(date)
   if (typeof date === 'string' && !/Z$/i.test(date)) {
     const d = date.match(C.REGEX_PARSE)
+
+    // Copied and modified from 'shye0000/dayjs'.
+    // https://github.com/iamkun/dayjs/pull/1862
     if (d) {
+      let dObj = null
+      const y = d[1]
       const m = d[2] - 1 || 0
       const ms = (d[7] || '0').substring(0, 3)
+      const dParams = [y, m, d[3] || 1, d[4] || 0, d[5] || 0, d[6] || 0, ms]
       if (utc) {
-        return new Date(Date.UTC(d[1], m, d[3]
-          || 1, d[4] || 0, d[5] || 0, d[6] || 0, ms))
+        dObj = new Date(Date.UTC(...dParams))
+      } else {
+        dObj = new Date(...dParams)
       }
```

Comment:
[nits]

How about using a ternary operator to avoid `let` declaration?

```js
const dObj = utc ? new Date(Date.UTC(...dParams)) : new Date(...dParams)
```